salt/_roster/kubernetes_nodes.py

```diff
@@ -0,0 +1,55 @@
+import fnmatch
+import logging
+log = logging.getLogger(__name__)
+
+try:
+    import kubernetes.client
+    import kubernetes.config
+    HAS_DEPS = True
+except ImportError:
+    HAS_DEPS = False
+
+__virtualname__ = 'kubernetes'
+
+def __virtual__():
+    if HAS_DEPS:
+        return HAS_DEPS
+    else:
+        return False, 'Unable to load the kubernetes nodes roster: dependencies unavailable.'
+
+
+def targets(tgt, tgt_type='glob', **kwargs):
+    if tgt_type != 'glob':
+        log.error('Only "glob" lookups are supported for now')
+        return {}
+
+    client = None
+    try:
+        client = kubernetes.config.new_client_from_config(
+            config_file='/etc/kubernetes/admin.conf',
+        )
+    except:
+        log.exception('Failed to load kubeconfig')
+        raise
+
+    v1 = kubernetes.client.CoreV1Api(api_client=client)
+
+    nodes = None
+    try:
+        nodes = v1.list_node()
+    except:
+        log.exception('Failed to retrieve v1/NodeList')
+        raise
+
+    # TODO Use `tgt_type`
+    return {
+        item.metadata.name: {
+            # Assume node name is resolvable
+            'host': item.metadata.annotations.get('metalk8s.scality.com/ssh-host', item.metadata.name),
+            'port': int(item.metadata.annotations.get('metalk8s.scality.com/ssh-port', 22)),
+            'user': item.metadata.annotations.get('metalk8s.scality.com/ssh-user', 'root'),
+            'priv': item.metadata.annotations.get('metalk8s.scality.com/ssh-priv-key-path', 'salt-ssh.rsa'),
```

Comment:
Lines are way too long…

You could declare an intermediate variable `annotations = item.metadata.annotations` and use that instead (yeah, you may have to drop to a `loop` instead of a dict comprehension but the readability gain is worth it IMHO).

Reply:
Done.